common/persistence/visibility/factory.go

```diff
@@ -282,6 +297,15 @@ func newVisibilityStoreFromDataStoreConfig(
 			metricsHandler,
 			logger,
 		)
+	} else if dsConfig.CustomDataStoreConfig != nil {
+		visStore, err = customVisibilityStoreFactory.NewVisibilityStore(
+			*dsConfig.CustomDataStoreConfig,
+			persistenceResolver,
+			logger,
+			metricsHandler,
+		)
+	} else {
+		return nil, nil
```

Comment:
`visStore` and `err` must be `nil` if it didn't enter any of those branches, so the last return should work.